src/QMCHamiltonians/QMCHamiltonian.cpp

```diff
@@ -488,19 +488,19 @@ QMCHamiltonian::Return_t QMCHamiltonian::evaluate(ParticleSet& P)
   return LocalEnergy;
 }
 
-void QMCHamiltonian::flex_evaluate(const RefVector<QMCHamiltonian>& H_list,
-                                   const RefVector<ParticleSet>& P_list,
-                                   std::vector<Return_t>& LocalEnergies)
+std::vector<QMCHamiltonian::RealType> QMCHamiltonian::flex_evaluate(const RefVector<QMCHamiltonian>& H_list,
+                                   const RefVector<ParticleSet>& P_list)
```

Comment:
This is fine for now, although I don't quite like the long calls.
The H object internally store LocalEnergy and KE and a few other breakups. I think we can just drop the return in the future and query H.LocalEnergy (of course protected by access functions.) I actually see it below.

Reply:
Why do we need two copies of the localEnergy. Id like to see it passed through the function chain until it reaches where it needs to rest. I think of it like the Hamiltonian is a list operators to apply and get a result. We want to have to be careful of the current state of as few objects as possible.

Reply:
There can be multiple consumers of the local energy. We can't repeated evaluate when we need the value. I think that is why final results rest inside H.